### PR TITLE
chore: Update dependencies 9/22

### PR DESCRIPTION
**Pull request by CelticMajora**
### 📰 Summary of changes
Updates the following:
- dart sdk to 2.18.1
- build_runner to 2.2.1
- freezed to 2.1.0
- json_serializable to 6.3.2
- lints to 2.0.0
- test to 1.21.6
- args to 2.3.1
- freezed_annotation to 2.1.0
- json_annotation to 4.6.0
- pub_semver to 2.1.1
- xml to 6.1.0

### 🧪 Testing done
Kept with existing tests to verify output was unchanged

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
         run: ./tool/generate_code_coverage.sh
 
       - name: Report code coverage
-        uses: zgosalvez/github-actions-report-lcov@v1.4.0
+        uses: zgosalvez/github-actions-report-lcov@v1.5.0
```
Comment (btrautmann) [inline .github/workflows/ci.yml:41]: Bonus points if we get this bad boy on Codecov! I think this is the last OSS project (in the mobile realm) not yet using it 👀 Could be wrong though...

**Review by btrautmann — APPROVED** [review]
domain lgtm

File: .github/workflows/ci.yml (hunk 1 above)
Comment (CelticMajora) [reply .github/workflows/ci.yml:41]: Can do in follow up PR

**Pull request merged**